core/src/Cache/TokenCacheSerializeHelper.cs

```diff
@@ -50,6 +50,8 @@ internal class TokenCacheSerializeHelper
         /// <param name="requestContext">call state to pass correlation id and logger instance</param>
         internal static void DeserializeUnifiedCache(TokenCacheAccessor tokenCacheAccessor, byte[] unifiedState, RequestContext requestContext)
         {
+            tokenCacheAccessor.Clear();
```

Comment:
>tokenCacheAccessor [](start = 12, length = 18)

This seems important (an alternative is to throw if the tokenAccessor is not empty, or to simply create the tokenCacheAccessor in the helper method).... are unit tests covering this helper's behavior?

Reply:
Why would you throw here ?

Reply:
I'm suggesting 2 ways to deal with this differently, to identify the better of these options or potentially a third I would need to understand why you had to add this in the first place.

1) To me it's magic that we decide to clear the object, and if we do there is really no need to pass it in. We can just return a new accessor instead. 
2) Alternatively if it's very important that the object has been cleared in advance then we can throw an exception instead (should the object not be in the expected state coming in)

How did you discover that .clear was needed?

Reply:
For custom cache persistence scenarios,  every cache access preceded by before access delegate call,  which is the place were cache supposed to be read from some persistent storage and deserialized to in-memory representation.
This is important to reflect current state of the cache, persistent state of cache  is "golden source" of it , not in-memory version.
For example : 
1.  you share cache between few app
2.  in  app1 you create msalClient1, then acquire token (now you have tokens in the cache).
3.  switch to app2 create msalClient2 then clear cache
4. return to app1  and execute getAccount on  msalClient1 - during deserialization   in memory representation of the cache should be cleared and instead persistent state (which is empty now) should be used,